index 71a5b5c..021be25 100644
--- src/Client.cpp --------------------------------
+++ src/Client.cpp --------------------------------
@@ -116,7 +116,7 @@ const double kSecs_to_usecs = 1e6;
 const int    kBytes_to_Bits = 8; 
 
 void Client::RunTCP( void ) {
-    unsigned long currLen = 0; 
+    long currLen; 
     struct itimerval it;
     max_size_t totLen = 0;
 
@@ -171,7 +171,7 @@ void Client::RunTCP( void ) {
 
         if ( !mMode_Time ) {
             /* mAmount may be unsigned, so don't let it underflow! */
-            if( mSettings->mAmount >= currLen ) {
+            if( mSettings->mAmount >= (unsigned long) currLen ) {
                 mSettings->mAmount -= currLen;
             } else {
                 mSettings->mAmount = 0;
