diff -uNr include/Settings.hpp include/Settings.hpp
--- include/Settings.hpp	2008-03-19 18:23:26.000000000 -0400
+++ include/Settings.hpp	2013-04-16 22:50:59.160600039 -0400
@@ -155,6 +155,7 @@
     // Hopefully int64_t's
     max_size_t mUDPRate;            // -b or -u
     max_size_t mAmount;             // -n or -t
+    max_size_t mBurstRate;          // -E
     // doubles
     double mInterval;               // -i
     // shorts
diff -uNr src/Client.cpp src/Client.cpp
--- src/Client.cpp	2010-04-01 16:23:17.000000000 -0400
+++ src/Client.cpp	2013-04-16 22:49:58.203599496 -0400
@@ -261,6 +261,7 @@
     reportstruct->packetID = 0;
 
     lastPacketTime.setnow();
+    unsigned int burstcount = 0;
     
     do {
 
@@ -310,10 +311,22 @@
         // report packets 
         reportstruct->packetLen = currLen;
         ReportPacket( mSettings->reporthdr, reportstruct );
-        
-        if ( delay > 0 ) {
+                
+        if (mSettings->mBurstRate)
+            {
+                if ((++burstcount >= mSettings->mBurstRate) && (delay > 0))
+                 {
+                     delay_loop( delay * mSettings->mBurstRate); 
+                 }
+                burstcount %= mSettings->mBurstRate;
+            }
+        else if (delay > 0)
+        {
             delay_loop( delay ); 
         }
+        
+        
+
         if ( !mMode_Time ) {
             /* mAmount may be unsigned, so don't let it underflow! */
             if( mSettings->mAmount >= currLen ) {
diff -uNr src/Settings.cpp src/Settings.cpp
--- src/Settings.cpp	2010-07-08 21:05:46.000000000 -0400
+++ src/Settings.cpp	2013-04-16 22:50:10.874600356 -0400
@@ -115,6 +115,7 @@
 {"listenport", required_argument, NULL, 'L'},
 {"parallel",   required_argument, NULL, 'P'},
 {"remove",           no_argument, NULL, 'R'},
+{"burstrate",  required_argument, NULL, 'E'},
 {"tos",        required_argument, NULL, 'S'},
 {"ttl",        required_argument, NULL, 'T'},
 {"single_udp",       no_argument, NULL, 'U'},
@@ -158,6 +159,7 @@
 {"IPERF_NODELAY",          no_argument, NULL, 'N'},
 {"IPERF_LISTENPORT", required_argument, NULL, 'L'},
 {"IPERF_PARALLEL",   required_argument, NULL, 'P'},
+{"IPERF_BURSTRATE",  required_argument, NULL, 'E'},
 {"IPERF_TOS",        required_argument, NULL, 'S'},
 {"IPERF_TTL",        required_argument, NULL, 'T'},
 {"IPERF_SINGLE_UDP",       no_argument, NULL, 'U'},
@@ -169,7 +171,7 @@
 
 #define SHORT_OPTIONS()
 
-const char short_options[] = "1b:c:df:hi:l:mn:o:p:rst:uvw:x:y:B:CDF:IL:M:N:O:P:RS:T:UVWZ:";
+const char short_options[] = "1b:c:df:hi:l:mn:o:p:rst:uvw:x:y:B:CDE:F:IL:M:N:O:P:RS:T:UVWZ:";
 
 /* -------------------------------------------------------------------
  * defaults
@@ -316,6 +318,9 @@
         case '1': // Single Client
             setSingleClient( mExtSettings );
             break;
+        case 'E':
+            mExtSettings->mBurstRate = atoi(optarg);
+            break;
         case 'b': // UDP bandwidth
             if ( !isUDP( mExtSettings ) ) {
                 fprintf( stderr, warn_implied_udp, option );
