diff -Naur src/Listener.cpp src/Listener.cpp
--- src/Listener.cpp	2014-07-02 12:29:32.323347129 -0400
+++ src/Listener.cpp	2014-07-03 15:38:09.473295780 -0400
@@ -675,40 +675,7 @@
  * --------------------------------------------------------------------*/ 
 
 void Listener::runAsDaemon(const char *pname, int facility) {
-#ifndef WIN32 
-    pid_t pid; 
-
-    /* Create a child process & if successful, exit from the parent process */ 
-    if ( (pid = fork()) == -1 ) {
-        fprintf( stderr, "error in first child create\n");     
-        exit(0); 
-    } else if ( pid != 0 ) {
-        exit(0); 
-    }
-
-    /* Try becoming the session leader, once the parent exits */
-    if ( setsid() == -1 ) {           /* Become the session leader */ 
-        fprintf( stderr, "Cannot change the session group leader\n"); 
-    } else {
-    } 
-    signal(SIGHUP,SIG_IGN); 
-
-
-    /* Now fork() and get released from the terminal */  
-    if ( (pid = fork()) == -1 ) {
-        fprintf( stderr, "error\n");   
-        exit(0); 
-    } else if ( pid != 0 ) {
-        exit(0); 
-    }
-
-    chdir("."); 
-    fprintf( stderr, "Running Iperf Server as a daemon\n"); 
-    fprintf( stderr, "The Iperf daemon process ID : %d\n",((int)getpid())); 
-    fflush(stderr); 
-
-    fclose(stdin); 
-#else 
+#ifdef WIN32 
     fprintf( stderr, "Use the precompiled windows version for service (daemon) option\n"); 
 #endif  
 
diff -Naur src/main.cpp src/main.cpp
--- src/main.cpp	2014-07-02 12:29:32.323347129 -0400
+++ src/main.cpp	2014-07-03 15:39:37.249294208 -0400
@@ -109,6 +109,47 @@
 // for all other threads to complete
 void waitUntilQuit( void );
 
+/* -------------------------------------------------------------------- 
+ * Run the server as a daemon  
+ * --------------------------------------------------------------------*/ 
+
+static void runAsDaemon( void ) {
+#ifndef WIN32
+    pid_t pid; 
+
+    /* Create a child process & if successful, exit from the parent process */ 
+    if ( (pid = fork()) == -1 ) {
+        fprintf( stderr, "error in first child create\n");     
+        exit(0); 
+    } else if ( pid != 0 ) {
+        exit(0); 
+    }
+
+    /* Try becoming the session leader, once the parent exits */
+    if ( setsid() == -1 ) {           /* Become the session leader */ 
+        fprintf( stderr, "Cannot change the session group leader\n"); 
+    } else {
+    } 
+    signal(SIGHUP,SIG_IGN); 
+
+
+    /* Now fork() and get released from the terminal */  
+    if ( (pid = fork()) == -1 ) {
+        fprintf( stderr, "error\n");   
+        exit(0); 
+    } else if ( pid != 0 ) {
+        exit(0); 
+    }
+
+    chdir("."); 
+    fprintf( stderr, "Running Iperf Server as a daemon\n"); 
+    fprintf( stderr, "The Iperf daemon process ID : %d\n",((int)getpid())); 
+    fflush(stderr); 
+
+    fclose(stdin);
+#endif
+}
+
 /* -------------------------------------------------------------------
  * main()
  *      Entry point into Iperf
@@ -188,6 +229,9 @@
             }
         }
 #endif
+        if ( isDaemon( ext_gSettings ) ) {
+            runAsDaemon();
+        }
         // initialize client(s)
         if ( ext_gSettings->mThreadMode == kMode_Client ) {
             client_init( ext_gSettings );
