--- src/SocketAddr.c	2007-08-29 23:57:27.000000000 +0200
+++ src/SocketAddr.c	2012-01-29 15:26:18.724669873 +0100
@@ -117,7 +117,12 @@
     }
         inSettings->size_local = sizeof( struct sockaddr_in );
 #endif
-    SockAddr_setPort( &inSettings->local, inSettings->mPort );
+    if ((inSettings->mLocalhost != NULL) && (inSettings->mHost != NULL))      
+        // enable -B -P, let the source port be chosen by the lower level
+	    // for each thread / connection
+	    SockAddr_setPort( &inSettings->local, 0 );
+    else
+        SockAddr_setPort( &inSettings->local, inSettings->mPort );
 }
 // end SocketAddr
 
