index b5d42b1..0f7bcb3 100644
--- src/Locale.c ---------------------------------
+++ src/Locale.c ---------------------------------
@@ -78,7 +78,7 @@ Usage: iperf [-s|-c host] [options]\n\
 Client/Server:\n\
   -f, --format    [kmKM]   format to report: Kbits, Mbits, KBytes, MBytes\n\
   -i, --interval  #        seconds between periodic bandwidth reports\n\
-  -l, --len       #[KM]    length of buffer to read or write (default 8 KB)\n\
+  -l, --len       #[KM]    length of buffer to read or write (default 128 KB)\n\
   -m, --print_mss          print TCP maximum segment size (MTU - TCP/IP header)\n\
   -o, --output    <filename> output the report or error message to this specified file\n\
   -p, --port      #        server port to listen on/connect to\n\

index 0107a63..912f210 100644
--- src/Settings.cpp -------------------------------
+++ src/Settings.cpp -------------------------------
@@ -202,7 +202,7 @@ void Settings_Initialize( thread_Settings *main ) {
     main->mFormat       = 'a';           // -f,  adaptive bits
     // skip help                         // -h,
     //main->mBufLenSet  = false;         // -l,	
-    main->mBufLen       = 128 * 1024;      // -l,  8 Kbyte
+    main->mBufLen       = 128 * 1024;      // -l,  128 Kbyte
     //main->mInterval     = 0;           // -i,  ie. no periodic bw reports
     //main->mPrintMSS   = false;         // -m,  don't print MSS
     // mAmount is time also              // -n,  N/A
